### ISSUE TITLE
Mac launch script doesn't work if any ancestor folders have space in name

### ISSUE THREAD

**Issue by JonFerraiolo** [maqetta/maqetta#3841]
I thought this worked in previous releases.

Should be easy fix - I would guess that we just need to properly quote variables.

I accidentally installed a downloadable zip in a path where there was a space in an ancestor folder name. (In this case, "maqetta 2"). The launch script died with this showing in my Terminal window:

```
Last login: Thu Mar 28 14:05:59 on ttys007
unknown14109fe36837:~ $ /Users/jonferraiolo/Downloads/maqetta\ 2/maqetta.server.mac.command ; exit;
Java version=1.7
/Users/jonferraiolo/Downloads/maqetta 2/maqetta.server.mac.command: line 60: test: /Users/jonferraiolo/Downloads/maqetta: binary operator expected
/Users/jonferraiolo/Downloads/maqetta 2/maqetta.conf not found
logout

[Process completed]
```

**Comment by jhpedemonte** [comment]
cc @peller

**Comment by JonFerraiolo** [comment]
The above commit is in branch 3841-LaunchScriptAllowSpaces. It includes various fixes that make the bash script work, but now I'm getting a "Error: Could not find or load main class 2.maqetta.conf" error if the current folder ends with the characters space and the 2 (i.e., `2`), and that error appears to be coming from Java.

The ultimate fix needs:
- Changes from above commit
- Migrate same changes to other UNIX launch scripts
- Figure out and fix why Java is generating the above message.

**Comment by JonFerraiolo** [comment]
Pushing to "future" because too difficult to fix for M10 and also because we have had the same problem with M7, M8, and M9 - basically, ever since we moved to Orion.

**Comment by jhpedemonte** [comment]
I did the same thing as that commit above.  However, what I found is that when embedding the double quotes in strings, Bash would add single quotes around the whole string when added elsewhere.  So for example, take this line:

```
JAVA_ARGS="${JAVA_ARGS} -Dorion.core.configFile=\"${MAQ_CONFIG}\""
```

When the command line is constructed using `JAVA_ARGS`, it looks something like this:

```
java 'JAVA_ARGS -Dorion.core.configFile="MAQ_CONFIG"' -other-option -etc-etc
```

Notice the single quotes.  Java wouldn't know how to parse that correctly.

**Comment by JonFerraiolo** [comment]
Where do you see the single quotes? When I put an `echo java ....` statement just before the `java ...` statement, I see this in the console:

```
java -Dorion.core.configFile="/Users/jonferraiolo/maqettaBuilds/maqetta-orion-201304191701 2/maqetta.conf" -Dmaqetta.siteConfigDirectory="/Users/jonferraiolo/maqettaBuilds/maqetta-orion-201304191701 2/siteConfig" -Dorg.eclipse.equinox.http.jetty.http.port=50000 -jar "/Users/jonferraiolo/maqettaBuilds/maqetta-orion-201304191701 2/plugins/org.eclipse.equinox.launcher_1.3.0.v20120522-1813.jar" -console -noExit -clean -data "/Users/jonferraiolo/maqettaBuilds/maqetta-orion-201304191701 2/users"
Error: Could not find or load main class 2.maqetta.conf"
```

In particular, notice that there aren't any single quotes.

**Comment by jhpedemonte** [comment]
I only added the escaped quotes around the config file declaration, around line 67.  Then, I ran the script with `bash -x`, which outputs the line that bash will run.  Saw this at the end:

```
+ echo 'Type "exit" or Ctrl-C to stop the server.'
Type "exit" or Ctrl-C to stop the server.
+ echo

+ '[' ']'
+ java '-Dorion.core.configFile="./maqetta.conf"' -Dmaqetta.siteConfigDirectory=./siteConfig -Dorg.eclipse.equinox.http.jetty.http.port=50000 -Dmaqetta.baseDirectory=/Users/pedemonte/builds/maqetta release 9/users -jar ./plugins/org.eclipse.equinox.launcher_1.3.0.v20120522-1813.jar -console -noExit -clean -data /Users/pedemonte/builds/maqetta release 9/users
Error: Could not find or load main class release
```

Notice the since quotes around the whole `configFile` bit.

**Comment by JonFerraiolo** [comment]
Thanks for the `bash -x` tip. That might help me.

**Comment by peller** [comment]
Found this similar Q&A on stack overflow.  Not sure I fully understand it yet.  Sounds like we may need arrays?

http://stackoverflow.com/questions/7502323/bash-quotes-in-debug-mode

**Comment by peller** [comment]
more stuff I don't completely understand.  "WordSplitting" seems relevant

http://stackoverflow.com/questions/4873083/bash-quoting-problem/4873727#4873727
http://mywiki.wooledge.org/BashFAQ/050

**Comment by JonFerraiolo** [comment]
Post mortem: After adding a bunch of double-quoting to the launch, I managed to get to the point where the entire script worked EXCEPT for the final "java" command. That command was failing because bash was ignoring double-quoted characters. We were passing in something like `java $JAVA_ARGS`, with double-quotes inside of the variable, which expanded into something like `java -Dfoo="aaa bbb" ...`, which then bash saw as 3 tokens: `java`, `-Dfoo="aaa` and `bbb"`. In other words, it looks like bash in this particular case was doing things in this order: (1) double-quote processing, (2) variable expansion, (3) wordsplitting, even though the docs claim things happen in a different order.

I then pounded my head against the wall, and did considerable research and experimentation yesterday for bash scripting around variable expansion and word splitting, and found explanations, but nothing that explained what I was seeing in practice. 

So, instead, I rewrote some of the logic in the launch script so as to avoid the expansion/splitting quirks that were causing things to break, and now things are working.
